xbmc/music/MusicUtils.cpp

```diff
@@ -9,384 +9,738 @@
+                                            labelMasks.m_strLabel2Folder);
+      for (const auto& i : items)
+      {
+        if (i->IsLabelPreformatted())
+          continue;
+
+        if (i->m_bIsFolder)
+          folderFormatter.FormatLabels(i.get());
+        else
+          fileFormatter.FormatLabels(i.get());
+      }
+
+      if (items.GetSortMethod() == SortByLabel)
+        items.ClearSortState();
 
-  std::vector<std::string> GetArtTypesToScan(const MediaType& mediaType)
+      items.Sort(state->GetSortMethod());
+    }
+
+    for (const auto& i : items)
+    {
+      GetItemsForPlaylist(i);
+    }
+  }
+  else
   {
-    std::vector<std::string> arttypes;
-    // Get default types of art that are to be automatically fetched during scanning
-    if (mediaType == MediaTypeArtist)
+    if (item->IsPlayList())
     {
-      arttypes = { "thumb", "fanart" };
-      for (auto& artType : CServiceBroker::GetSettingsComponent()->GetSettings()->GetList(
-        CSettings::SETTING_MUSICLIBRARY_ARTISTART_WHITELIST))
+      const std::unique_ptr<PLAYLIST::CPlayList> playList(
+          PLAYLIST::CPlayListFactory::Create(*item));
+      if (!playList)
       {
-        if (find(arttypes.begin(), arttypes.end(), artType.asString()) == arttypes.end())
-          arttypes.emplace_back(artType.asString());
+        CLog::Log(LOGERROR, "{} failed to create playlist {}", __FUNCTION__, item->GetPath());
+        return;
       }
+
+      if (!playList->Load(item->GetPath()))
+      {
+        CLog::Log(LOGERROR, "{} failed to load playlist {}", __FUNCTION__, item->GetPath());
+        return;
+      }
+
+      for (int i = 0; i < playList->size(); ++i)
+      {
+        GetItemsForPlaylist((*playList)[i]);
+      }
+    }
+    else if (item->IsInternetStream() && !item->IsMusicDb())
+    {
+      // just queue the internet stream, it will be expanded on play
+      m_queuedItems.Add(item);
     }
-    else if (mediaType == MediaTypeAlbum)
+    else if (item->IsPlugin() && item->GetProperty("isplayable").asBoolean())
     {
-      arttypes = { "thumb" };
-      for (auto& artType :
-        CServiceBroker::GetSettingsComponent()->GetSettings()->GetList(
-          CSettings::SETTING_MUSICLIBRARY_ALBUMART_WHITELIST))
+      // python files can be played
+      m_queuedItems.Add(item);
+    }
+    else if (!item->IsNFO() && (item->IsAudio() || item->IsVideo()))
+    {
+      const auto itemCheck = m_queuedItems.Get(item->GetPath());
+      if (!itemCheck || itemCheck->GetStartOffset() != item->GetStartOffset())
       {
-        if (find(arttypes.begin(), arttypes.end(), artType.asString()) == arttypes.end())
-          arttypes.emplace_back(artType.asString());
+        // add item
+        m_musicDatabase.SetPropertiesForFileItem(*item);
+        m_queuedItems.Add(item);
       }
     }
-    return arttypes;
   }
+}
+
+void ShowToastNotification(const CFileItem& item, int titleId)
```

Comment:
Another one that doesn't belong here, not your fault

Reply:
That's the question. Looking at the content of this source file I consider it "GUI".

Reply:
Looking at the atual usages of the include I'd say it's debatable. Used mainly from GUI components but also from application and infoscanners...

Reply:
Call hierarchy shows that `ShowToastNotification`is purely GUI and not revealed in the header.

<img width="372" alt="Screenshot 2022-10-26 at 08 51 09" src="https://user-images.githubusercontent.com/3226626/197955186-2b8492a8-48d8-4a71-8d5f-ebb69f000ea6.png">

I agree that MusicUtils.h could be refactored into a GUI and a non-GUI part, but this is for another day (PR).